Actron air add sensor entity type

kclif9 opened the pull request:
## Proposed change
Adds the sensor entity type to the Actron Air integration. The majority of the A/C sensors are marked as disabled diagnostic sensors and can be enabled by the user if they wish. The only enabled out of the box one is clean filter. 

Each peripheral (A/C temperature sensor) is zone assignable, but may have multiple sensors per zone. These are represented as individual objects. 

## Type of change
- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [X] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/42662/
- Link to developer documentation pull request: 
- Link to frontend pull request: 

## Checklist

- [X] I understand the code I am submitting and can explain how it works.
- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [X] I have followed the [perfect PR recommendations][perfect-pr]
- [X] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [X] Tests have been added to verify that the new code works.
- [X] Any generated code has been carefully reviewed for correctness and compliance with project standards.

If user exposed functionality or configuration variables are added/changed:

- [X] Documentation added/updated for [www.home-assistant.io][docs-repository]
https://github.com/home-assistant/home-assistant.io/pull/42662/

If the code communicates with devices, web services, or third-party tools:

- [X] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [X] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [X] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

kclif9 committed:
Add switch entity platform

kclif9 committed:
Merge branch 'dev' into Actron-Air-Add-Switch-Entity-Type

kclif9 committed:
fix copilot suggestions

kclif9 committed:
wip

kclif9 committed:
wip

kclif9 committed:
wip

kclif9 committed:
fix state class

kclif9 committed:
mark completed items as done

kclif9 committed:
update units of measurement

kclif9 committed:
fix state classes

kclif9 committed:
Merge branch 'dev' into Actron-Air-Add-Sensor-Entity-Type

zweckj commented on homeassistant/components/actron_air/quality_scale.yaml:
```diff
@@ -26,7 +26,9 @@ rules:
-  action-exceptions: todo
+  action-exceptions:
+    status: exempt
+    comment: This integration does not have custom service actions.
```
this also applies to the actions in switches, so this is incorrect

zweckj commented on homeassistant/components/actron_air/sensor.py:
```diff
@@ -0,0 +1,326 @@
+        return not self.coordinator.is_device_stale()
+
+
+class AirconCleanFilterSensor(BaseAirconSensor):
```
instead of defining a lot of custom classes we rather use `EntityDescriptions` (in this case `SensorEntityDescription`) which we just loop over. Look at some other integrations for examples e.g. `lamarzocco`. It's fine to keep two different classes for peripheral types, but we can greatly reduce the amount of code here

kclif9 committed:
Refactor Actron Air integration: remove switch platform, add sensor entities, and update entity names

kclif9 committed:
update PR with recent changes to avoid merge conflicts

kclif9 committed:
fix what will be merge conflicts when #158087 is merged

kclif9 committed:
Merge branch 'dev' into Actron-Air-Add-Sensor-Entity-Type

kclif9 committed:
Refactor Actron Air sensor tests to validate peripheral sensor keys and improve assertions

kclif9 committed:
Update quality_scale.yaml to mark entity categories, device classes, and disabled status as done

kclif9 committed:
Update quality_scale.yaml to mark action-exceptions and reconfiguration-flow as exempt, and set entity-translations and exception-translations to done

kclif9 committed:
Refactor Actron Air integration:
- Update climate entities to inherit from ActronAirEntity.
- Modify coordinator to use ActronAirStatus.
- Enhance sensor entity descriptions and implement value functions.
- Adjust quality scale to reflect action-exceptions as todo.
- Improve test cases for sensor entities and peripheral sensors.

kclif9 committed:
Merge branch 'dev' into Actron-Air-Add-Sensor-Entity-Type
